### PR TITLE
Update Modbus component configuration variable

### PR DESCRIPTION
**Pull request by klaasnicolaas**
Update style of Modbus component (all the 4 files) documentation to follow new configuration variables description.
Related to #6385.

**Description:**

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** home-assistant/home-assistant#<home-assistant PR number goes here>

## Checklist:

- [ ] Branch: `next` is for changes and new documentation that will go public with the next [home-assistant](https://github.com/home-assistant/home-assistant) release. Fixes, changes and adjustments for the current release should be created against `current`.
- [ ] The documentation follow the [standards][standards].

[standards]: https://home-assistant.io/developers/documentation/standards/

### CODE REVIEW

**Comment by klaasnicolaas** [comment]
Error: Liquid Exception: (<unknown>): mapping values are not allowed in this context at line 3 column 69 in /opt/build/repo/source/_components/modbus.markdown

File: source/_components/binary_sensor.modbus.markdown [hunk 1]
```diff
@@ -32,13 +32,26 @@ binary_sensor:
         coil: 110
 ```
 
-Configuration variables:
+{% configuration %}
+coils:
+  description: The array contains a list of coils to read from.
+  required: true
+  type: map
```
Comment (WofWca) [inline source/_components/binary_sensor.modbus.markdown]: List maybe? Or [map, list]? Idk.

File: source/_components/modbus.markdown [hunk 2]
```diff
@@ -30,15 +30,32 @@ modbus:
...
-- **type** (*Required*): Type of the connection to Modbus. Possible values are:
-  - *tcp*: Modbus TCP protocol according to "MODBUS Messaging Implementation Guide version 1.0b" provided by Schneider Automation, 
-  - *udp*: Modbus TCP form, but using UDP for transport (removes the overheads required for TCP),
-  - *rtuovertcp*: Modbus RTU message transmitted with a TCP/IP wrapper and sent over a network instead of serial lines.
-- **host** (*Required*): The IP address of your Modbus device, eg. 192.168.1.1.
-- **port** (*Required*): The port for the communication.
-- **timeout** (*Optional*): Timeout for slave response in seconds. (default: 3)
+{% configuration %}
+type:
+  description: Type of the connection to Modbus. Possible values are:
```
Comment (WofWca) [inline source/_components/modbus.markdown]: `:`

File: source/_components/sensor.modbus.markdown [hunk 3]
```diff
@@ -45,21 +45,66 @@ sensor:
...
+      required: false
+      default: 0
+      type: integer
+    data_type:
+      description: Response representation (int, uint, float, custom). If float selected, value will be converted to IEEE 754 floating point format.
+      required: false
+      default: int
+      type: string
+    structure:
+      description: If data_type is custom specify here a double quoted python struct format string to unpack the value. See python documentation for details. Ex: ">i".
```
Comment (WofWca) [inline source/_components/sensor.modbus.markdown]: Here

File: source/_components/switch.modbus.markdown [hunk 4]
```diff
@@ -40,20 +40,72 @@ switch:
...
+      required: false
+      default: True
+      type: boolean
+    verify_register:
+      description: Register to readback.
+      required: false
+      default: same as register
+      type: string
+    register_type:
+      description: Modbus register type: holding or input.
```
Comment (WofWca) [inline source/_components/switch.modbus.markdown]: And here.

**Comment by klaasnicolaas** [comment]
if it is good, it can now be merged

File: source/_components/modbus.markdown [hunk 5]
```diff
@@ -30,15 +30,32 @@ modbus:
...
-  - *udp*: Modbus TCP form, but using UDP for transport (removes the overheads required for TCP),
-  - *rtuovertcp*: Modbus RTU message transmitted with a TCP/IP wrapper and sent over a network instead of serial lines.
-- **host** (*Required*): The IP address of your Modbus device, eg. 192.168.1.1.
-- **port** (*Required*): The port for the communication.
-- **timeout** (*Optional*): Timeout for slave response in seconds. (default: 3)
+{% configuration %}
+type:
+  description: Type of the connection to Modbus. Possible values are `tcp`, `udp` and `rtuovertcp`.
+  required: true
+  type: map
```
Comment (frenck) [inline source/_components/modbus.markdown]: This is not a map or list. This is a string.

**Review by frenck — CHANGES_REQUESTED** [review]
@klaasnicolaas Found 1 little thing. Could you take a look? 👍

**Review by frenck — APPROVED** [review]

**Comment by frenck** [comment]
Thanks, @klaasnicolaas!

Happy Hacktoberfest 🎉 

../Frenck

**Pull request merged**